presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkNativeTaskExecutorFactory.java

```diff
@@ -0,0 +1,494 @@
+        this.authenticatorProviders = ImmutableSet.copyOf(requireNonNull(authenticatorProviders, "authenticatorProviders is null"));
+        this.nativeExecutionProcessFactory = requireNonNull(nativeExecutionProcessFactory, "processFactory is null");
+        this.nativeExecutionTaskFactory = requireNonNull(nativeExecutionTaskFactory, "taskFactory is null");
+        this.shuffleInfoTranslator = requireNonNull(shuffleInfoTranslator, "shuffleInfoFactory is null");
+    }
+
+    @Override
+    public <T extends PrestoSparkTaskOutput> IPrestoSparkTaskExecutor<T> create(
+            int partitionId,
+            int attemptNumber,
+            SerializedPrestoSparkTaskDescriptor serializedTaskDescriptor,
+            Iterator<SerializedPrestoSparkTaskSource> serializedTaskSources,
+            PrestoSparkTaskInputs inputs,
+            CollectionAccumulator<SerializedTaskInfo> taskInfoCollector,
+            CollectionAccumulator<PrestoSparkShuffleStats> shuffleStatsCollector,
+            Class<T> outputType)
+    {
+        try {
+            return doCreate(
+                    partitionId,
+                    attemptNumber,
+                    serializedTaskDescriptor,
+                    serializedTaskSources,
+                    inputs,
+                    taskInfoCollector,
+                    shuffleStatsCollector,
+                    outputType);
+        }
+        catch (RuntimeException e) {
+            throw executionExceptionFactory.toPrestoSparkExecutionException(e);
+        }
+    }
+
+    public <T extends PrestoSparkTaskOutput> IPrestoSparkTaskExecutor<T> doCreate(
+            int partitionId,
+            int attemptNumber,
+            SerializedPrestoSparkTaskDescriptor serializedTaskDescriptor,
+            Iterator<SerializedPrestoSparkTaskSource> serializedTaskSources,
+            PrestoSparkTaskInputs inputs,
+            CollectionAccumulator<SerializedTaskInfo> taskInfoCollector,
+            CollectionAccumulator<PrestoSparkShuffleStats> shuffleStatsCollector,
+            Class<T> outputType)
+    {
+        PrestoSparkTaskDescriptor taskDescriptor = taskDescriptorJsonCodec.fromJson(serializedTaskDescriptor.getBytes());
+        ImmutableMap.Builder<String, TokenAuthenticator> extraAuthenticators = ImmutableMap.builder();
+        authenticatorProviders.forEach(provider -> extraAuthenticators.putAll(provider.getTokenAuthenticators()));
+
+        Session session = taskDescriptor.getSession().toSession(
+                sessionPropertyManager,
+                taskDescriptor.getExtraCredentials(),
+                extraAuthenticators.build());
+        PlanFragment fragment = taskDescriptor.getFragment();
+        StageId stageId = new StageId(session.getQueryId(), fragment.getId().getId());
+        TaskId taskId = new TaskId(new StageExecutionId(stageId, 0), partitionId, attemptNumber);
+
+        // TODO: Remove this once we can display the plan on Spark UI.
```

Comment:
This is quite unfortunate. But, why do we need every executor to log plan fragment? Shouldn't we have driver log these once?

Reply:
PlanFragment log in executor aids debugging to easily spot which fragment is being executed. Also, given internally at Meta, executors logs always contains logs of ALL the tasks run on that executor v/s just that task, it is hard to visually draw boundaries on these logs without plan fragment logging.